scripts/test.py

```diff
@@ -0,0 +1,146 @@
+                            action='version', version=str(build_script.VERSION))
+    arg_parser.add_argument(
+        '--test-exe', dest='test_exe', metavar='TEST_EXE',
+        action='store', default=None,
+        help='Test executable to run')
+    arg_parser.add_argument(
+        '--test-args', dest='test_args', metavar='TEST_ARGS', nargs='+',
+        action='store',
+        help='Test arguments passed to the test executable')
+    arg_parser.add_argument(
+        '--build-dir', dest='build_dir', metavar='BUILD_DIR',
+        action='store', default=None)
+    arg_parser.add_argument(
+        '-a', '--arch', dest='architecture',
+        metavar='ARCH', action='store', choices=build_script.ARCHITECTURES,
+        default=build_script.DEFAULT_ARCHITECTURE,
+        help='Build target architecture. Can be one of: {0}'.format(
+                ', '.join(build_script.ARCHITECTURES)))
+    arg_parser.add_argument(
+        '-c', '--config', dest='configuration',
+        metavar='CONFIG', action='store', choices=build_script.CONFIGURATIONS,
+        default=build_script.DEFAULT_CONFIGURATION,
+        help='Build target configuration. Can be one of: {0}'.format(
+            ', '.join(build_script.CONFIGURATIONS)))
+    return arg_parser.parse_args()
+
+
+def run_test(test_exe, test_args):
+    '''
+    Run a single test
+    '''
+    run_test_args = [test_exe]
+    if args.test_args is not None:
+        run_test_args.extend(args.test_args)
+    run_test_work_dir = os.path.split(test_exe)[0]
+    if '' == run_test_work_dir:
+        run_test_work_dir = os.getcwd()
+    try:
+        test_result = subprocess.run(run_test_args, cwd=run_test_work_dir)
+        if 0 != test_result.returncode:
+            raise TestError('Running test', ' '.join(run_test_args), 'failed')
+    except Exception:
+        raise TestError('Failed to run test', test_exe, ' '.join(test_args))
+
+
+# Main entry point
+if '__main__' == __name__:
+    try:
+        build_script = import_build_script()
+        args = parse_args(build_script)
+        tests = []
+        if args.test_exe is None:
+            for test in ALL_TESTS.items():
+                test_exe_dir = args.build_dir
+                if test_exe_dir is None:
+                    test_exe_dir = os.path.join(
+                        build_script.BUILD_CONFIGS[args.configuration],
+                        platform.system().lower(),
+                        args.architecture,
+                        'bin')
+                test_exe = os.path.join(test_exe_dir, test[0])
+                test_args = copy.deepcopy(test[1])
+                if args.test_args is not None:
+                    test_args.extend(args.test_args)
+                tests.append((test_exe, test_args))
+        else:
+            tests = [(args.test_exe, args.test_args)]
+        for test_exe, test_args in tests:
+            run_test(test_exe, test_args)
+    except Exception as error:
+        print('Error', *(error.args))
```

Comment:
This loses information about the exception that caused the error, preventing a debugging engineer from seeing the stack trace.  Is there a good reason?  **Not** catching the exception at all will allow the normal Python error handling, which will include a non-zero exit code and a helpful stack trace.

Reply:
This provides a nice output for QA and QE - the plan is for AMD QA/QE run the same test script.
I'll update this to include the exception name, but the feedback we got from QA is that the stack trace is of no interest to them.

Reply:
QA includes not only driver QA, but test harness QA.  I understand why a test harness problem is of no interest to driver QA.  But it is of interest to whoever is maintaining the harness.  Without the normal stack trace, some classes of problems (particularly problems that seem to appear randomly) are difficult to debug.

Reply:
How would prefer this to be handled?
1. Completely remove the try/except?
2. Add a printout of the callstack in except?